common/src/main/java/com/wynntils/models/players/PlayerModel.java

```diff
@@ -87,15 +88,40 @@ public boolean isLocalPlayer(String name) {
 
     public boolean isNpc(Player player) {
         StyledText scoreboardName = StyledText.fromString(player.getScoreboardName());
-        return isNpcName(scoreboardName) || isNpcUuid(player.getUUID());
+        return isNpcName(scoreboardName);
+    }
+
+    public boolean isDisplayPlayer(Player player) {
+        // This is a "fake" player, like an NPC but that you can't interact with,
+        // like the player at the character selection screen
+        var uuid = player.getUUID();
+        if (uuid.version() == 4) return false;
+        if (isNpc(player)) return false;
+        if (isPlayerGhost(player)) return false;
+
+        return true;
     }
 
     public boolean isPlayerGhost(Player player) {
         return ghosts.containsKey(player.getUUID());
     }
 
-    public WynntilsUser getUser(UUID uuid) {
-        return users.getOrDefault(uuid, null);
+    public UUID getUserUUID(Player player) {
+        var uuid = player.getUUID();
+        if (uuid == null) return null;
+
+        if (uuid.version() == 4) {
+            // This is a local player
+            return uuid;
+        } else {
+            // Ghost players have their UUIDs converted to version 2; convert it back
+            UUID uuidV4 = new UUID((uuid.getMostSignificantBits() & ~0xF000L) | 0x4000L, uuid.getLeastSignificantBits());
+            return uuidV4;
+        }
```

Comment:
Is this a bug I wonder :)

<img width="725" height="503" alt="Image" src="https://github.com/user-attachments/assets/d78eb290-a103-4afb-a16f-0ec2993e4faf" />